### PR TITLE
Fix controller-gen and operator-sdk versions

### PR DESCRIPTION
**Pull request by jzding**
- Update versions of controller-gen and operator-sdk.
- Add/fix the installation of controller-gen and operator-sdk in Makefile.
- Remove crd options trivialVersions and preserveUnknownFields since they are no longer supported in later version of controller-gen.

Fix the following issue:
### crd options trivialVersions and preserveUnknownFields are obsolete
```
$ make generate && make manifests
workspace/openshift/ptp-operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
workspace/openshift/ptp-operator/bin/controller-gen "crd:trivialVersions=true,preserveUnknownFields=false" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
Error: unable to parse option "crd:trivialVersions=true,preserveUnknownFields=false": [unknown argument "trivialVersions" (at <input>:1:16) extra arguments provided: "true,preserveUnknownFields=false" (at <input>:1:17)]
```
### could not find manager deployment, should have label control-plane=controller-manager 
ref: https://github.com/operator-framework/operator-sdk/issues/5574
This issue is fixed in operator-sdk v1.19.1, therefore upgrade the operator-sdk to v1.22.0
```
$ make bundle
workspace/openshift/ptp-operator/bin/controller-gen crd rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
workspace/openshift/ptp-operator/bin/operator-sdk generate kustomize manifests --interactive=false -q
cd config/manager && /home/jacding/workspace/openshift/ptp-operator/bin/kustomize edit set image controller=quay.io/openshift/origin-ptp-operator:4.12
workspace/openshift/ptp-operator/bin/kustomize build config/manifests | workspace/openshift/ptp-operator/bin/operator-sdk generate bundle -q --overwrite --version 4.12.0   --extra-service-accounts "linuxptp-daemon"
FATA[0000] Error generating bundle manifests: could not find manager deployment, should have label control-plane=controller-manager 
make: *** [Makefile:161: bundle] Error 1

```
Signed-off-by: Jack Ding <jackding@gmail.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -130,6 +128,14 @@ CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
 controller-gen: ## Download controller-gen locally if necessary.
 	$(call go-install-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.9.2)
 
+OPERATOR_SDK = $(shell pwd)/bin/operator-sdk
+OPERATOR_SDK_VERSION = $(shell $(OPERATOR_SDK) version 2>/dev/null | sed 's/^operator-sdk version: "\([^"]*\).*/\1/')
+OPERATOR_SDK_VERSION_REQ = v1.22.0-ocp
+operator-sdk: ## Download operator-sdk locally if necessary.
+ifneq ($(OPERATOR_SDK_VERSION_REQ),$(OPERATOR_SDK_VERSION))
+	curl https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/operator-sdk/4.11.0/operator-sdk-v1.22.0-ocp-linux-x86_64.tar.gz | tar -xz -C bin/
```
Comment (josephdrichard) [inline Makefile:136]: Can you specify OPERATOR_SDK_VERSION_REQ here?
Or just use https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/operator-sdk/4.11.0/operator-sdk-linux-x86_64.tar.gz?

Comment (jzding) [reply Makefile]: updated the package name

**Review by aneeshkp — APPROVED** [review]

**Pull request merged**